src/views/product-integrations-landing/contexts/integrations-search-context.tsx

```diff
@@ -76,52 +85,80 @@ export const IntegrationsSearchProvider: React.FC<Props> = ({
 		sharedOptions
 	)
 
-	const officialChecked = qsTiers.includes(Tier.OFFICIAL)
-	const partnerChecked = qsTiers.includes(Tier.PARTNER)
-	const communityChecked = qsTiers.includes(Tier.COMMUNITY)
-
-	const setOfficialChecked = (value: boolean) => {
-		if (value) {
-			setQsTiers((prev) => [...prev, Tier.OFFICIAL])
-		} else {
-			setQsTiers((prev) => prev.filter((tier) => tier !== Tier.OFFICIAL))
-		}
-	}
-	const setPartnerChecked = (value: boolean) => {
-		if (value) {
-			setQsTiers((prev) => [...prev, Tier.PARTNER])
-		} else {
-			setQsTiers((prev) => prev.filter((tier) => tier !== Tier.PARTNER))
-		}
-	}
-	const setCommunityChecked = (value: boolean) => {
-		if (value) {
-			setQsTiers((prev) => [...prev, Tier.COMMUNITY])
-		} else {
-			setQsTiers((prev) => prev.filter((tier) => tier !== Tier.COMMUNITY))
-		}
-	}
-
 	// Filter out integrations that don't have releases yet
 	const integrations = useMemo(() => {
 		return _integrations.filter((integration: Integration) => {
 			return integration.versions.length > 0
 		})
 	}, [_integrations])
 
-	const flags: Flag[] = integrations
-		// accumulate all integration flags
-		.flatMap((e) => e.flags)
-		// remove duplicates
-		.filter((value, index, self) => {
-			const _value = JSON.stringify(value)
-			return (
-				index ===
-				self.findIndex((obj) => {
-					return JSON.stringify(obj) === _value
+	// accumulate all unique filterable integration fields
+	// - tiers
+	// - components
+	// - flags
+	// - types
+	const { tiers, components, flags, types } = integrations.reduce(
+		(acc, next) => {
+			// collect tiers
+			if (!acc.tiers.some((e) => e == next.tier)) {
+				acc.tiers = acc.tiers.concat(next.tier)
+				acc.tiers = acc.tiers.sort((a: Tier, b: Tier) => {
+					if (tierSortVal(a) > tierSortVal(b)) {
+						return 1
+					} else if (tierSortVal(a) < tierSortVal(b)) {
+						return -1
+					} else {
+						return 0
+					}
 				})
-			)
+			}
+			// collect components
+			next.components.forEach((component) => {
+				if (!acc.components.some((e) => e.id == component.id)) {
+					acc.components = acc.components.concat(component).sort((a, b) => {
+						const textA = a.name.toLowerCase()
+						const textB = b.name.toLowerCase()
+						return textA < textB ? -1 : textA > textB ? 1 : 0
+					})
+				}
+			})
+			// collect flags
+			next.flags.forEach((flag) => {
+				if (!acc.flags.some((e) => e.id == flag.id)) {
+					acc.flags = acc.flags.concat(flag)
+				}
+			})
+			// collect types
+			// TODO(kevinwang): add support for types
+			return acc
+		},
+		{
+			tiers: [] as Tier[],
+			components: [] as IntegrationComponent[],
+			flags: [] as Flag[],
+			types: [] as $TSFixMe[], // TODO(kevinwang): add support for types
+		}
+	)
+
+	const tiersCheckedArray = useMemo(() => {
```

Comment:
the previous `officialChecked`, `partnerChecked`, `communityChecked` vars are now replaced by `tiersCheckedArray`, resulting in a more uniform pattern.